syntax = "proto3";
package service;




import "container_sdk/model/container/resource_group.proto";
import "container_sdk/model/container/workload.proto";
import "container_sdk/model/container/service.proto";

/*
Get请求
*/
message GetRequest {
    /*
    获取指定的 Service id
    */
    string instanceId = 1 ;
}

/*
Get返回
*/
message GetResponse {
    /*
    关联的部署资源组 (resource group)
    */
    container.ResourceGroup resourceGroup = 1 ;
    /*
    关联的 Workload
    */
    container.Workload workload = 2 ;
    /*
    Service 信息
    */
    container.Service service = 3 ;
}

/*
GetApi返回
*/
message GetResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    service.GetResponse data = 4 ;
}

